analyses/cell-type-dsrct/exploratory_analysis/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,441 @@
+
+```{r}
+# classify tumor cells based on presence of any marker genes 
+marker_sum_exp <- lapply(marker_sum_exp, function(x){
+  x |> 
+  dplyr::mutate(sum_classification = dplyr::if_else(sum_exp > 0, "Tumor", "Normal"))})
+
+
+for(i in 1:length(marker_sum_exp)) {
+  p <- ggplot(marker_sum_exp[[i]], aes(x = UMAP1, y = UMAP2, color = sum_classification)) +
+  geom_point(size = 0.5, alpha = 1) +
+  ggtitle(names(umap_df_list_excluded_scaled)[i])
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      module_base,
+      '/plots/UMAP_classification_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+}
+
+```
+
+This gives us a rough idea of cells that may be classified as tumor cells. However, I believe re-doing this analysis using the cluster level data may give us better results. This is highly dependent on whether the clusters were generated with enough granularity.
+
+```{r}
+# calculate sum gene expression across all marker genes in list 
+marker_sum_exp <- lapply(umap_df_list_excluded_scaled, function(x){
+  x |> 
+  dplyr::group_by(cluster) |> #change to cluster
+  dplyr::mutate(sum_exp = sum(transformed_gene_expression, na.rm = T)) |> 
+  dplyr::select(barcodes, UMAP1, UMAP2, sum_exp, cluster) |> 
+  dplyr::distinct()
+})
+
+
+# plot mean gene expression 
+
+for(i in 1:length(marker_sum_exp)) {
+  p <- ggplot(marker_sum_exp[[i]], aes(x = UMAP1, y = UMAP2, color = sum_exp)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  scale_color_viridis_c() +
+    ggtitle(names(umap_df_list_excluded_scaled)[i])
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(module_base, '/plots/UMAP_marker_expression_cluster_', names(umap_df_list)[i], '.png'),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+}
+
+```
+It looks like there are group of cells with high marker gene expression. We  anticipate that these are most likely to be the tumor cells. The groups with low or negative values are likely normal cells. 
+
+Now let's classify clusters that has a sum of marker genes > 0 (after z-transformation) as tumor cell clusters. 
+
+
+```{r}
+# classify tumor cells based on presence of any marker genes 
+marker_sum_exp <- lapply(marker_sum_exp, function(x){
+  d <- density(x$sum_exp)
+  
+  x |> 
+  dplyr::mutate(sum_classification = dplyr::if_else(sum_exp > 0, "Tumor", "Normal"))})
+
+
+for(i in 1:length(marker_sum_exp)) {
+  p <- ggplot(marker_sum_exp[[i]], aes(x = UMAP1, y = UMAP2, color = sum_classification)) +
+  geom_point(size = 0.5, alpha = 1) +
+  ggtitle(names(umap_df_list_excluded_scaled)[i])
+  print(p)
+
+  #save plots
+  ggsave(
+    filename = paste0(
+      module_base,
+<<<<<<< HEAD
+      '/plots/UMAP_classification_cluster_',
+=======
+      '../plots/UMAP_classification_cluster_',
+>>>>>>> danhtruong/danhtruong/cell-type-dsrct
```

Comment:
Danh: there are a number of places where it seems like your latest changes have introduced merge artifacts like this one. Can you please fix these and rerender the notebook? I would expect to see a newer version of the html that matches the latest updates.

Reply:
Apologies... on closer inspection it looks like you did rerun the notebook, but then a later merge brought in the above error (which should have been caught by our pre-commit tests, but wasn't for some reason!). I have reverted the offending commit and merged in the changes from the `main` repository, so this should be ready to go, but you will need to pull down the changes from this branch (or start a new branch from `main` after this is merged in) before proceeding with your next steps.